tests/core/nogo/custom/custom_test.go

```diff
@@ -280,6 +291,48 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	return nil, nil
 }
 
+-- flagger.go --
+// flagger crashes when three flags are set in the config or else it no-ops
+package flagger
+
+import (
+        "errors"
+        "fmt"
+
+        "golang.org/x/tools/go/analysis"
+)
+
+var (
+        boolSwitch   bool
+        stringSwitch string
+        intSwitch    int
+)
+
+var Analyzer = &analysis.Analyzer{
+        Name: "flagger",
+        Run:  run,
+        Doc:  "Dummy analyzer that requires some flags to be set true in order to pass",
+}
+
+func init() {
+        Analyzer.Flags.BoolVar(&boolSwitch, "bool-switch", false, "Bool must be set to true to run")
+        Analyzer.Flags.StringVar(&stringSwitch, "string-switch", "no", "String must be set to \"yes\" to run")
+        Analyzer.Flags.IntVar(&intSwitch, "int-switch", 0, "Int must be set to 1 to run")
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+        if !boolSwitch {
+                return nil, nil
+        }
+        if stringSwitch != "yes" {
+                return nil, nil
+        }
+        if intSwitch != 1 {
+                return nil, nil
+        }
+        return nil, errors.New("all switches were set -> fail")
+}
```

Comment:
Nit: Since we are allowing users to provide the FlagSet as a `map[string]string`, it might worth providing a test to document how nogo would behave when user were to pass some data that is not the correct data type.

For example: Let's have a test where user provided `int-switch: one` to an IntVar flag and expect some error?

Reply:
Ah, I just realized `FlagSet.Set` actually returns an error on failure to parse 🤦 . Let me see what I can do here.

Reply:
Okay, partly surfacing errors from `FlagSet.Set`. I'm not including the full error text since it's usually not very helpful, but I'm at least surfacing whether the flag is not defined, invalid, or erroneously provided with a leading hyphen. Added a `analyzer_flags` section in the .rst for the nogo config file as well.

Reply:
@sluongng Looks like @NavneethJayendran (AKA me) addressed your review. Mind taking another look?